dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ToolTesterResource.java

```diff
@@ -0,0 +1,105 @@
+/*
+ *
+ *  *    Copyright 2019 OICR
+ *  *
+ *  *    Licensed under the Apache License, Version 2.0 (the "License");
+ *  *    you may not use this file except in compliance with the License.
+ *  *    You may obtain a copy of the License at
+ *  *
+ *  *        http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  *    Unless required by applicable law or agreed to in writing, software
+ *  *    distributed under the License is distributed on an "AS IS" BASIS,
+ *  *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *    See the License for the specific language governing permissions and
+ *  *    limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.resources;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.codahale.metrics.annotation.Timed;
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.DockstoreWebserviceConfiguration;
+import io.dockstore.webservice.core.ToolTester.ToolTesterLog;
+import io.dockstore.webservice.core.ToolTester.ToolTesterLogType;
+import io.dockstore.webservice.core.ToolTester.ToolTesterS3Client;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.apache.http.HttpStatus;
+
+/**
+ * @author gluu
+ * @since 1.7.0
+ */
+@Path("/toolTester")
+@Produces(MediaType.APPLICATION_JSON)
+@Tag(name = "toolTester", description = "Interactions with the Dockstore-support's ToolTester application")
+public class ToolTesterResource {
+    private final String bucketName;
+
+    public ToolTesterResource(DockstoreWebserviceConfiguration configuration) {
+        bucketName = configuration.getToolTesterBucket();
+    }
+
+    @GET
+    @Timed
+    @Path("logs")
+    @Operation(summary = "Get ToolTester log file")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getToolTesterLog(
+            @QueryParam("tool_id") @Parameter(example = "#workflow/github.com/dockstore/hello_world", required = true) String toolId,
```

Comment:
Maybe it would be overly verbose, but also have an example for a tool? I think the `#workflow` thing is a subtlety that new users may not know off-hand.